genny/auth/templates/models/user.go.plush

```diff
@@ -31,15 +35,48 @@ type User struct {
 // Create wraps up the pattern of encrypting the password and
 // running validations. Useful when writing tests.
 func (u *User) Create(tx *pop.Connection) (*validate.Errors, error) {
-	u.Email = strings.ToLower(u.Email)
-	ph, err := bcrypt.GenerateFromPassword([]byte(u.Password), bcrypt.DefaultCost)
+	verrs := u.validatePassword()
+	if verrs.HasAny() {
+		return verrs, nil
+	}
+
+	ph, err := encryptPassword(u.Password)
 	if err != nil {
 		return validate.NewErrors(), errors.WithStack(err)
 	}
 	u.PasswordHash = string(ph)
+	u = u.sanitizeFields()
 	return tx.ValidateAndCreate(u)
 }
 
+// Update handles the extra work possibly needed during user update,
+// hashing password and making email lowercase for consistency
+func (u *User) Update(tx *pop.Connection) (*validate.Errors, error) {
+	if len(u.Password) != 0 {
```

Comment:
This check should not be neccessary, `u.validatePassword()` also checks the length

Reply:
This is a check to see if they are updating their password or not. If so we validate it and then update it, otherwise we sanitize and then update the rest of the model.